src/components/pages/ImageClassification/ImageAnnotationModal/ImageAnnotationModalMap.js

```diff
@@ -224,45 +263,99 @@ const ImageAnnotationModalMap = ({
 
     const handleMapLoad = () => {
       setHasMapLoaded(true)
+
       map.current.loadImage('/cross-hair.png', (error, image) => {
         if (error) {
           return
         }
-        map.current.addImage('crosshair', image)
+        map.current.addImage('cross-hair', image)
         map.current.addLayer({
           id: 'patches-center-layer',
           type: 'symbol',
           source: 'patches-center',
           layout: {
-            'icon-image': 'crosshair',
+            'icon-image': 'cross-hair',
           },
         })
       })
+      map.current.loadImage('/label-background.png', (error, image) => {
+        if (error) {
+          return
+        }
+
+        map.current.addImage('label-background', image, {
+          // this configuration allows the image to stretch around the label text
+          stretchX: [[5, 135]],
+          stretchY: [[5, 135]],
+          content: [5, 5, 135, 135],
+          pixelRatio: 2,
+        })
+      })
     }
-    const displayPointFeatureLabel = ({ features }) => {
-      const [{ geometry, properties }] = features
-      map.current.getCanvas().style.cursor = 'pointer'
-      const label = properties.isUnclassified ? 'Unclassified' : properties.ba_gr_label
-      pointLabelPopup.setLngLat(geometry.coordinates[0][0]).setHTML(label).addTo(map.current)
-    }
-    const hidePointFeatureLabel = () => {
-      map.current.getCanvas().style.cursor = ''
-      pointLabelPopup.remove()
-    }
+
     map.current.on('load', handleMapLoad)
-    map.current.on('mouseenter', 'patches-fill-layer', displayPointFeatureLabel)
-    map.current.on('mouseleave', 'patches-fill-layer', hidePointFeatureLabel)
 
     const currentMap = map.current
     return () => {
       currentMap.off('load', handleMapLoad)
-      currentMap.off('mouseenter', 'patches-fill-layer', displayPointFeatureLabel)
-      currentMap.off('mouseleave', 'patches-fill-layer', hidePointFeatureLabel)
       currentMap.remove()
     }
     // eslint-disable-next-line
   }, [])
```

Comment:
two things:
- for dep arrays, we shoudlnt ignore linting 99.99% of the time. However fixing this opens a can of worms large enough that its easy to argue that its out of scope for this work. 
- effects should be single purpose, so its good to remove the management of the hover popup to its own effect, but also I had no choice since the popup stuff required me to add stuff the the dep array, and if I kept it in the original effect and messed with its dep array, it broke a bunch of things.